enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,374 @@
+2. **Components-Controller**: This controller reconciles the same `LoggingStack` resource into the resources `ClusterLogging`, `LoggingStack` and `ObservabilityUIPlugin` that in turn install and manage the logging stack components (e.g. collection, storage and visualization plugin).
+
+### LoggingStack status conditions
+
+To effectively track the success/failure status at any point in time of the managed operators (i.e. CLO and LO), the following conditions will be aggregated from OLM resources into the `LoggingStack` status conditions:
+1. TBD
+2. ...
+
+To effectively track the success/failure status of the managed but shared custom resources, the following conditions will be aggregated from the `ClusterLogging` and `LokiStack` resources into the `LoggingStack` status conditions:
+1. TBD
+2. ...
+
+Finally the following conditions will be managed in the `LoggingStack` status conditions about visulization and correlation:
+1. TBD
+2. ...
+
+### Automated korrel8r deployment management
+
+A separate controller namely `correlation_controller` will be introduced in the Cluster Obserbability Operator watching only on create and delete events of `MonitoringStack` and `LoggingStack` resources. If both exist the [korrel8r](korrel8r) deployment will be fully automated on the Cluster Observability Operator installation namespace.
+
+#### Automated korrel8r deployment with CMO/UWM
+
+For OpenShift environments still depending on Cluster-Monitoring-Operator managing the monitoring aspects of infrastructure and workloads (i.e. via User Workload Monitoring) the event of any Alertmanager instance being used on this platform can be be used as a surrogate event for a missing `MonitoringStack`. The events to track Alertmanager instances (via existence of Kubernetes `Service` resources) is a robust approach to inject [korrel8r](korrel8r) as per now the UI links are shown in Alerting related pages only.
+
+### Risks and Mitigations
+
+1. **Risk**: Tracking the installation/upgrading path following the trail of multiple Operator Framework resources (`Subscription` -> `InstallPlan` -> `ClusterServiceVersion`) requires detailed due dilligence on how to categorize and summarize their conditions into high-level ones for a better user experience.
+2. **Risk**: We are well advised to consider the long-term goal of the Operator Framework to migrate to a newly multi-tenant / multi-version operator management approach using new designed CRDs which introduce breaking changes.
+3. **Risk**: To manage install/uninstall robustly the Cluster Observability Operator is required to become aware of CRD registration at each time which in turn requires to re-instatiate internals like informers/listers/etc. This might require considerable custom design and work as per controller-runtime abstraction not designed or allowing restarting entire controllers.
```

Comment:
(nit)
```suggestion
3. **Risk**: To manage install/uninstall robustly the Cluster Observability Operator is required to become aware of CRD registration at each time which in turn requires to re-instantiate internals like informers/listers/etc. This might require considerable custom design and work as per controller-runtime abstraction not designed or allowing restarting entire controllers.
```

Reply:
Regarding this, a crude approach could be for COO to restart any time it detects that an "interesting" CRD is created/deleted?